Add AI Notebooks Troubleshooting documentation (#2914)
ovh/docs: pages/platform/ai/notebook_guide_troubleshooting/guide.en-gb.md

---
title: AI Notebooks - Troubleshooting
slug: notebook/debug-notebooks
excerpt: Tutorial about how to debug your notebooks
section: AI Notebooks guides
order: 05
---

**Last updated 14th of April, 2022.**

## Objective

This tutorial gives you some hints on how to debug your notebooks if things go wrong.

## Requirements

- a notebook you would like to start with **AI Notebooks**

> - a **PCS** account

Reviewer: peut être ne pas utiliser l'acronyme pour être plus clair ?

Reviewer 2: suggestion: 
- an [Object Storage Container](https://docs.ovh.com/gb/en/storage/pcs/create-container/) in your OVHcloud account

Author: By habit, and in the other docs, it is noted PCS. I don't know if it's better to rename it...

- a Public Cloud user with the **Administrator** or **AI Training Operator** role

## I can't connect to my notebook

If you are unable to connect to AI Notebooks, please ensure the following:

> - **Your notebook is up and running:** to connect to your notebook, it must be in running or in service status (green colored). If your notebook is starting up, wait a few moments until you data and you will be able to connect. Starting time may vary due due to some parameters. The more data your synchronize, the more time it will take.

Reviewer: ```suggestion
- **Your notebook is up and running:** to connect to your notebook, it must be in running or in service status (green colored). If your notebook is starting up, wait a few moments until your data and you will be able to connect. Starting time may vary due  to some parameters. The more data your synchronize, the more time it will take.
```

Pas sûr de la phrase ` wait a few moments until your data and you will be able to connect`, notamment le `your data`

- **You have an autorized user or an authorized token:** to be able to connect to your notebook, you must have created a user or token. To create a new user, log into the OVHcloud Control Panel and open your Public Cloud project. Click on `Users & Roles` in the left menu under `Project Management`. In the roles edition, don't forget to select `Administrator` and/or `AI Training Operator` for your user.
To follow all the steps in detail, please refer to this [documentation](https://docs.ovh.com/gb/en/publiccloud/ai/users/).

> As an alternative, you can generate an authorized token directly via our CLI.

Reviewer: peut être ajouter un lien vers la doc qui explique comment faire ?

1. `pip` command:

``` {.console}
pip install package-name
```

2. `conda` command:

``` {.console}
conda install package-name
```

- **Install packages in a terminal:** First, open a new terminal.
If you have chosen the `Jupyterlab` editor, open a new terminal: `file` > `new` > `terminal`
If you have chosen the `VScode` editor, open a new terminal: `application menu` > `terminal` > `new terminal`

You can then execute:

``` {.console}
pip install package-name
```

or

``` {.console}
conda install package-name
```

directly in your terminal.

- **Install packages directly in a notebook cell:** If you want to install a package directly in a notebook cell, do not forget to put an exclamation mark in front of your command. Then, you will be able to install a package with:

``` {.console}
!pip install package-name
```

or

``` {.console}
!conda install package-name
```

It is still advisable to install or uninstall packages in a terminal, particularly for authorizations. You can authorize (or not) the installation or uninstallation of packages with a yes or no. It is impossible to do this in a notebook cell.

## I don't see my data synchronized in the Object Storage

> When you create an AI Notebook, we copy you data pushed in attached Object Storage containers near you compute resources (`CPU` and `GPU`) to increase performance and reduce latency.

Reviewer: ```suggestion
When you create an AI Notebook, we copy your data pushed in attached Object Storage containers near your compute resources (`CPU` and `GPU`) to increase performance and reduce latency.
```

> [!primary]
>
> When you write data inside an attached volume, we wait for the AI Notebook stop to synchronize back your data.
>

If you are unable to see your data synchronized, please ensure the following:

> - **Your notebook is stopped:** To see your synchronized data in the Object Storage, your notebook must be stopped beforehand.The status of your notebook must be stopped (red colored). If your notebook is `STOPPING`, wait a few moments and you will be able to see your synchronized data.

Reviewer: ```suggestion
- **Your notebook is stopped:** To see your synchronized data in the Object Storage, your notebook must be stopped beforehand. The status of your notebook must be stopped (red colored). If your notebook is `STOPPING`, wait a few moments and you will be able to see your synchronized data.
```

- **Your object container is not empty:** If your object container is empty, you may not have saved your data in the right place when using it in your notebook. Check how the volume was mounted when you started your notebook.

## I'm unable to install the CLI (Command Line Interface)

To install the CLI for OVHcloud AI Tools, you can check this and follow the different steps.

For informations, two regions are available: `bhs` and `gra`.

- **Your cluster location is `gra`:** Go to this [page](https://cli.gra.training.ai.cloud.ovh.net/)

- **Your cluster location is `bhs`:** Go to this [page](https://cli.bhs.training.ai.cloud.ovh.net/)

You have to choose the zip file corresponding to your operating system. Then, unzip your file by running the following command in a terminal:

``` {.console}
unzip ovhai-<platform>.zip
```

> [!primary]
>
> <platform> is the name of your operating system. It corresponds to the one indicated in the zip file you downloaded.
>

> Now all you have to do is adding it to your `PATH`:

Reviewer: ```suggestion
Now, all you have to do is to extract it and add it to your `PATH`:
```
